speechbrain/lobes/models/huggingface_transformers/huggingface.py

```diff
@@ -0,0 +1,404 @@
+            )
+            self.freeze_model(self.model)
+        else:
+            self.model.gradient_checkpointing_disable()  # Required by DDP
+            self.model.train()
+
+    def _from_pretrained(
+        self, source, config, model, save_path, cache_dir,
+    ):
+        """This function manages the source checking and loading of the params.
+        # 1. Is the model from HF or a local path
+        # 2. Is the model pretrained with HF or SpeechBrain
+        # 3. Download (if appropriate) and load with respect to 1. and 2.
+        Arguments
+        ---------
+        source : str
+            HuggingFace hub name: e.g "facebook/wav2vec2-large-lv60"
+        config : AutoConfig
+            HuggingFace generic configuration class.
+        model: AutoModel
+            HuggingFace generic model class.
+        save_path : str
+            Path (dir) of the downloaded model.
+        cache_dir : str
+            Path (dir) in which a downloaded pretrained model configuration should be cached.
+        """
+        is_sb, ckpt_file, is_local = self._check_model_source(source, save_path)
+        if is_sb:
+            config = config.from_pretrained(source, cache_dir=save_path)
+            self.model = model(config)
+            self.model.gradient_checkpointing_disable()  # Required by DDP
+            # fetch the checkpoint file
+            ckpt_full_path = fetch(
+                filename=ckpt_file,
+                source=source,
+                savedir=save_path,
+                cache_dir=cache_dir,
+            )
+            # We transfer the parameters from the checkpoint.
+            self._load_sb_pretrained_parameters(
+                ckpt_full_path, modify_state_dict_fn=self._modify_state_dict,
+            )
+        else:
+            if self.for_pretraining:
+                # For now, we don't need to load pretrained model for pretraining
+                # To be modified in the future to support more complicated scenerios
+                # For example fine-tuning in the SSL manner
+                self.model = model
+            else:
+                self.model = model.from_pretrained(
+                    source, config=config, cache_dir=save_path
+                )
+
+    def _check_model_source(self, path, save_path):
+        """Checks if the pretrained model has been trained with SpeechBrain and
+        is hosted locally or on a HuggingFace hub.
+        Called as static function in HFTransformersInterface._from_pretrained.
+        Arguments
+        ---------
+        path : str
+            Used as "source"; local path or HuggingFace hub name: e.g "facebook/wav2vec2-large-lv60"
+        save_path : str
+            norm_output (dir) of the downloaded model.
+        Returns
+        -------
+        is_sb : bool
+            Whether/not the model is deserializable w/ SpeechBrain or not (then, model conversion is needed).
+        checkpoint_filename : str
```

Comment:
We are returning `return is_sb, checkpoint_filename, is_local`, please update the return docstring accordingly . We are missing the `is_local` variable